parser/internal/parservX/parser.go

```diff
@@ -818,6 +825,146 @@ func (p *parser) parse() (ast.Expression, error) {
 	return result, nil
 }
 
+func (p *parser) parseFunctionDefinitons() error {
+	functions := make(map[string]*ast.FunctionDefinition)
+
+	for {
+		tok, lit, _ := p.scanIgnoreWhitespace()
+		if tok == ast.Name && lit == "def" {
+			function, err := p.parseFunctionDefinition()
+			if err != nil {
+				return err
+			}
+			functions[function.Name] = function
+		} else {
+			p.unscan()
+			break
+		}
+	}
+
+	p.functions = functions
```

Comment:
Can you add a test where we assert that functions are picked up? Since we skip those lines it's not part of the ast